lazy/lazy.go

```diff
@@ -0,0 +1,206 @@
+package lazy
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/daotl/go-datastore"
+	"github.com/daotl/go-datastore/key"
+	"github.com/daotl/go-datastore/query"
+)
+
+var (
+	ErrClosed = errors.New("datastore closed")
+)
+
+type DSModifier func(d *datastore.Datastore) error
+
+// LazyDatastore wraps a datastore with three states: active, inactive and closed.
+// LazyDatasotre is inited by four DSModifier: init, activate, deactive and closeFunc.
```

Comment:
Unify naming:
// LazyDatasotre is inited by four DSModifier: initFn, activateFn, deactiveFn, and closeFn.